src/components/TodoModal/TodoModal.tsx

```
@@ -1,42 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppSelector } from '../../app/hooks';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const todo = useAppSelector(status => status.currentTodo);
+  const userId = todo?.userId || 0;
+  const setTodo = currentTodoSlice.actions.setCurrentTodo;
+  const [user, setUser] = useState<User>();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getUser(userId)
+      .then(userFromServer => setUser(userFromServer))
+      .catch()
```

The `.catch()` block is empty, so any errors from the `getUser()` promise will be ignored. It's better to at least log the error or handle it appropriately to avoid silent failures and make debugging easier.